src/sentry_plugins/asana/plugin.py

```diff
@@ -0,0 +1,251 @@
+from __future__ import absolute_import
+
+import six
+
+from rest_framework.response import Response
+
+from sentry.exceptions import InvalidIdentity, PluginError, PluginIdentityRequired
+from sentry.plugins.bases.issue2 import IssuePlugin2, IssueGroupActionEndpoint
+from sentry.utils.http import absolute_uri
+
+from sentry_plugins.base import CorePluginMixin
+from sentry_plugins.exceptions import ApiError, ApiUnauthorized
+from .client import AsanaClient
+
+
+ERR_INTERNAL = (
+    'An internal error occurred with the integration and '
+    'the Sentry team has been notified'
+)
+
+ERR_UNAUTHORIZED = (
+    'Unauthorized: either your access token was invalid or '
+    'you do not have access'
+)
+
+ERR_AUTH_NOT_CONFIGURED = (
+    'You still need to associate an Asana identity with this account.'
+)
+
+
+class AsanaPlugin(CorePluginMixin, IssuePlugin2):
+    description = 'Integrate Asana issues by linking a repository to a project.'
+    slug = 'asana'
+    title = 'Asana'
+    conf_title = title
+    conf_key = 'asana'
+    auth_provider = 'asana'
+
+    def get_group_urls(self):
+        return super(AsanaPlugin, self).get_group_urls() + [
+            (r'^autocomplete', IssueGroupActionEndpoint.as_view(
+                view_method_name='view_autocomplete',
+                plugin=self,
+            )),
+        ]
+
+    def is_configured(self, request, project, **kwargs):
+        return bool(self.get_option('workspace', project))
+
+    def has_workspace_access(self, workspace, choices):
+        return workspace in set([c[0] for c in choices])
```

Comment:
Pro Python tip: Set's have a shorter syntax here.

```python
{c[0] for c in choices}
```

But alternatively, you're creating a set, just to check if something is contained in it. If the goal is to be more efficient, you can write it as:

```python
for c, _ in choices:
  if workspace == c:
    return True
return False
```